go/sql/user_device.go

```diff
@@ -104,14 +104,14 @@ func (e *UserDeviceJournal) Fields() []string {
 // UserDeviceSyncRecord is the joined row of UserDevice and UserDeviceJournal
 type UserDeviceSyncRecord struct {
 	UserDeviceJournal
-	State string `json:"state" gorm:"column:state"`
+	State *string `json:"state" gorm:"column:state"`
 }
 
 // Row
 func (r *UserDeviceSyncRecord) Row() SyncRow {
 	return &UserDevice{
 		Key:   r.UserDeviceJournal.Key,
-		State: r.State,
+		State: *r.State,
```

Comment:
I'm not sure about this part by the way, because it seems that we call SyncRecord.Row() also on deletion sync records? That wouldn't work.

Reply:
You can set State only if r.State is not null.  Then the default state will be empty string. Or also make UserDevice.State a pointer?